sync-core/src/main/java/com/cloudant/sync/query/Index.java

```diff
@@ -0,0 +1,165 @@
+     * @param indexSettings the optional settings used to configure the index.
+     *                      Only supported parameter is 'tokenize' for text indexes only.
+     * @return the Index object or null if arguments passed in were invalid.
+     */
+    public static Index setUp(List<Object> fieldNames,
+                              String indexName,
+                              String indexType,
+                              Map<String, String> indexSettings) {
+        if (fieldNames == null || fieldNames.isEmpty()) {
+            logger.log(Level.SEVERE, "No field names were provided.");
+            return null;
+        }
+
+        if (indexName == null || indexName.isEmpty()) {
+            logger.log(Level.SEVERE, "No index name was provided.");
+            return null;
+        }
+
+        if (indexType == null || indexType.isEmpty()) {
+            logger.log(Level.SEVERE, "No index type was provided.");
+            return null;
+        }
+
+        if (!validTypes.contains(indexType.toLowerCase())) {
+            logger.log(Level.SEVERE, String.format("Invalid index type %s.", indexType));
+            return null;
+        }
+
+        if (indexType.equalsIgnoreCase(JSON_TYPE) && indexSettings != null) {
+            logger.log(Level.WARNING, String.format("Index type is %s, index settings %s ignored.",
+                                                    indexType,
+                                                    indexSettings.toString()));
+            indexSettings = null;
+        } else if (indexType.equalsIgnoreCase(TEXT_TYPE)) {
+            if (indexSettings == null) {
+                indexSettings = new HashMap<String, String>();
+                indexSettings.put(TOKENIZE, DEFAULT_TOKENIZER);
+                logger.log(Level.INFO, String.format("Index type is %s, defaulting settings to %s.",
+                        indexType,
+                        indexSettings.toString()));
+            } else {
+                for (String parameter : indexSettings.keySet()) {
+                    if (!validSettings.contains(parameter.toLowerCase())) {
+                        String msg = String.format("Invalid parameter %s in index settings %s.",
+                                                   parameter,
+                                                   indexSettings);
+                        logger.log(Level.SEVERE, msg);
+                        return null;
+                    }
+                }
+            }
+        }
+
+        return new Index(fieldNames, indexName, indexType, indexSettings);
+    }
+
+    /**
+     * Compares the index type and accompanying settings with the passed in arguments.
+     *
+     * @param indexType the index type to compare to
+     * @param indexSettings the indexSettings to compare to
+     * @return true/false - whether there is a match
+     */
+    protected boolean compareIndexTypeTo(String indexType, String indexSettings) {
```

Comment:
Right now this method provides a way (again) to declutter the IndexCreator logic that checks whether an index already exists or needs to be created.  It could have also been written as a method within the IndexCreator itself.  I think that as the Index class evolves it may wind up that we change it to implement Comparable.  But a valid point.